seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/SeleniumComJSRunner.scala

```diff
@@ -41,37 +40,37 @@ class SeleniumComJSRunner(browserProvider: SeleniumBrowser,
     awaitForBrowser(timeout)
     @tailrec def loop(): String = {
       val code = "return this.scalajsSeleniumComJSRunnerChannel.popOutMsg();"
-      browser.getWebDriver.executeScript(code) match {
+      driver.executeScript(code) match {
         case null =>
           loop()
 
         case msg: String =>
-          browser.processConsoleLogs(console)
+          processConsoleLogs(console)
           if (msg.startsWith(MESSAGE_TAG)) {
             val taglessMsg = msg.substring(MESSAGE_TAG.length)
             "&[0&]".r.replaceAllIn(taglessMsg, regMatch =>
                 if (regMatch.group(0) == "&&") "&" else "\u0000")
           } else if (msg == CLOSE_TAG) {
             throw new ComJSEnv.ComClosedException("Closed from browser.")
           } else {
-            BrowserDriver.illFormattedScriptResult(msg)
+            illFormattedScriptResult(msg)
           }
 
         case obj =>
           // Here we only try to get the console because it uses the same
           // communication channel that is potentially corrupted.
-          Try(browser.processConsoleLogs(console))
-          BrowserDriver.illFormattedScriptResult(obj)
+          Try(processConsoleLogs(console))
+          illFormattedScriptResult(obj)
       }
     }
     loop()
   }
 
-  def close(): Unit = {
-    browser.processConsoleLogs(console)
+  override def close(): Unit = {
+    processConsoleLogs(console)
     comClosed = true
-    if (!keepAlive || ignoreKeepAlive)
-      browser.close()
+    if (!config.keepAlive || ignoreKeepAlive)
+      super.close()
```

Comment:
That's cleaned up in the end state I believe. We specifically do not want to *never* call `_driver.close()` if `keepAlive` is set.